kola/tests/etcd/discovery.go

```diff
@@ -62,9 +64,41 @@ func discovery(cluster platform.Cluster, version int) error {
 		plog.Infof("etcd instance%d started", i)
 	}
 
-	err := getClusterHealth(cluster.Machines()[0], csize)
-	if err != nil {
-		return fmt.Errorf("discovery failed health check: %v", err)
+	if version == 2 {
+		err := getClusterHealth(cluster.Machines()[0], csize)
+		if err != nil {
+			return fmt.Errorf("discovery failed health check: %v", err)
+		}
+	} else if version == 1 {
+		var keyMap map[string]string
+		var retryFuncs []func() error
+
+		retryFuncs = append(retryFuncs, func() error {
+			var err error
+			keyMap, err = setKeys(cluster, 5)
+			if err != nil {
+				//return err
+				return fmt.Errorf("setkeys %v", err)
+			}
+			return nil
+		})
+		retryFuncs = append(retryFuncs, func() error {
+			if err := checkKeys(cluster, keyMap, false); err != nil {
+				//return err
+				return fmt.Errorf("checkkeys %v", err)
+			}
+			return nil
+		})
+		for _, retry := range retryFuncs {
+			if err := util.Retry(5, 5*time.Second, retry); err != nil {
+				return err
```

Comment:
might as well wrap this in the same error message that etcd2 discovery check would print for consistency:

`return fmt.Errorf("discovery failed health check: %v", err)`